### PR TITLE
Fix online check with HTTP proxy

### LINKED ISSUE #933: Network never detected, app does not start
**Describe the bug**
New feature "waiting for network" does not give expected result in my environment. The application never starts and always hangs on a blank page showing "no internet connection" in the title bar. I tried multiple times and cleared storage without success. 

Of course, I made sure network was available prior to filling this issue which is the case: I can browse any external site from my browser including the teams web site and downgrading the app to 1.3.3 works as expected for now.

**To Reproduce**
1. Start the app (either with or without clearing storage)
2. Watch the title bar showing "waiting for network"
3. Wait until the title bar changes to "no internet connection"
4. The app stays frozen on a blank page and does not start

**Expected behavior**
The app detect network correctly and starts normally

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04
 - Installation package deb (via apt channel)
 - Version 1.3.5

**Additional context**
Possibly interfering:
- I run a local squid proxy forwarding requests to my company proxy. It runs on a local IP on port 8080
- I have 2 network interfaces enabled on my machine (although leaving only one enabled did not change anything).

### PR DESCRIPTION
**Pull request by thomasgl-orange**
Fixes #933. This PR changes `httpHelper` to use the [Electron `net`](https://www.electronjs.org/docs/latest/api/net) module as a replacement for Node.js `http`/`https` modules. The `net` module offers transparent support for web proxies, whereas Node.js `http`/`https` module requires complicated code to correctly issue a request through a proxy.

With this change, the online check "works for me" (on startup and wakeup from sleep), both when directly connected to internet and when on a corporate network with mandatory use of a proxy for internet access. Note that the `net` module requires the application being ready before it becomes usable, so I've had to delay `downloadCustomBGServiceRemoteConfig()` a bit (in `app/index.js`, I've moved the call into the `ready` event handler).

This PR also removes the unused `checkConnectivity` module; this dead-code removal is not strictly necessary (it's just me tidying code when I started looking into this issue), so if you prefer a PR without this commit I can amend it.

### CODE REVIEW

**Comment by IsmaelMartinez** [comment]
Thanks for your contribution! I'm considering whether we should implement this change under a config option (feature flag). This approach would allow us to thoroughly test it in isolation before a full rollout at a later stage.

This pull request has an example for your reference: [Link to PR](https://github.com/IsmaelMartinez/teams-for-linux/pull/688/files).

The rationale behind this consideration is our substantial user base, which has exceeded 115,000 active users, only counting the SNAP active users. Implementing such a significant change could potentially impact a large number of Teams-for-Linux users, which we'd like to avoid if possible!

I also want to express my thanks for the numerous improvements and the removal of redundant code you've made. I really appreciate it. Thanks again!

**Comment by jijojosephk** [comment]
@thomasgl-orange Big thanks for your efforts! May be, the custom background config download can be kept as it is unless you find benefit. The proxy thing is a bit complicated. Have you checked `1.3.8` ? It was converted to `HTTP` from `DNS` lookup. So, the problem might be gone. But not sure it works in all proxy cases. I have no experience with proxy servers so can't comment on that. @IsmaelMartinez as you said, this can be added with a config option with `disabled` default so nothing gets broken. Again @thomasgl-orange much elated to see contributions. Keep doing it. Good day!

**Comment by thomasgl-orange** [comment]
Thanks @IsmaelMartinez and @jijojosephk. I've found a bit of time to add more commits in this PR, to make the online check thing configurable. Was that something like this you had in mind? I've added a `--checkOnlineMethod` option with this possible values:
- `https` (default): uses the `net` API to make an HTTPS request to the main URL (the one from `--url`, or `https://teams.microsoft.com` - I had previously hard-coded `https://teams.microsoft.com`, but maybe some companies use something else somehow and would block this default URL, who knows, so it's better this way I guess...)
- `dns`: uses the `net` API to make a DNS query on the hostname from the main URL (it's a variation of the old, pre-1.3.4, test); not reliable for some use-case involving proxy
- `native`: uses the Electron `net.isOnline()` method (works for me if directly connected to internet, but it's too optimistic to check that the mandatory proxy, if any, is ready for use)
- `none`: just assume we're online (that's more for testing, or unblocking situations we would not have anticipated)

The `isOnline()` code is moved in the `ConnectionManager`, it was easier to give it access to the config and logger this way, and it's only used there anyway.

A test release is available here: https://github.com/thomasgl-orange/teams-for-linux/releases/tag/v1.3.8-tgl3

> May be, the custom background config download can be kept as it is unless you find benefit.

I'm not using it personally, but I'm pretty sure it was broken for non-local URL in a "behind proxy" context; the Nodejs `http`/`https` API which were used in `httpHelper` require more code than there was to actually use a proxy (that's the benefit of Electron's `net` API). So I guess I've fixed something here.

But now that I've separated `isOnline()` from `httpHelper`, we could as well revert `httpHelper` back to using `http`/`https`. Let me know if you prefer that, if you feel it's safer.

> Have you checked 1.3.8 ? It was converted to HTTP from DNS lookup. So, the problem might be gone.

Actually, for me (mandatory proxy for internet access, but DNS lookup works), this change was a blocking regression (the online test always failing, whereas it was working before with DNS, although it was inaccurate because not a guarantee that proxy was available). That's what brought me here :)

**Comment by jijojosephk** [comment]
@IsmaelMartinez, @thomasgl-orange has put a lot into this and I find it very useful.

@thomasgl-orange, try to see if can fix the `codebeat` issues so that code will be much cleaner. I can see `codebeat Pending — 1 issue resolved and 2 introduced`. Also, the `package.json` and `com.github.IsmaelMartinez.teams_for_linux.appdata.xml` files require version updates. `1.3.10`. If you can have them updated I would request @IsmaelMartinez to review once again and merge.

Fantastic work @thomasgl-orange ! :+1:

**Comment by jijojosephk** [comment]
Also I suggest to keep the default `none` so that no one will be affected by an unforeseen bug.

**Comment by thomasgl-orange** [comment]
Hi @jijojosephk,

> try to see if can fix the `codebeat` issues

Sure, I had overlooked that. I've split the long `isOnline` method, `codebeat` is happy now.

> Also, the `package.json`` and com.github.IsmaelMartinez.teams_for_linux.appdata.xml` files require version updates. `1.3.10`.

Sure, I can do that, but `1.3.10` or `1.3.9`? (can't find 1.3.9 in develop)

> Also I suggest to keep the default `none` so that no one will be affected by an unforeseen bug.

I think it's better to keep `https`. Switching to `none` by default would, for sure, be a regression for some users, where waiting for network is actually required for the app to properly start (or wakeup). Whereas `https` _may_ lead to some regression (new code can have bugs, etc.), but it's only a possibility. If that happens, `--onlineCheckMethod` will be available as a workaround, until the hypothetical bug gets fixed.

**Comment by thomasgl-orange** [comment]
> Sure, I can do that, but 1.3.10 or 1.3.9? (can't find 1.3.9 in develop)

Okay, I see the 1.3.9 in `package.json` now. Still, since there's no tag or release yet, I'll assume I can continue with this version number (and include @IsmaelMartinez's latest change in the appdata.xml changelog). I'll commit that, and will amend if that's not okay.

File: com.github.IsmaelMartinez.teams_for_linux.appdata.xml [hunk 1]
```diff
@@ -14,6 +14,15 @@
 	<url type="bugtracker">https://github.com/IsmaelMartinez/teams-for-linux/issues</url>
 	<launchable type="desktop-id">com.github.IsmaelMartinez.teams_for_linux.desktop</launchable>
 	<releases>
+		<release version="1.3.9" date="2023-09-07">
```
Comment (jijojosephk) [inline com.github.IsmaelMartinez.teams_for_linux.appdata.xml]: Version is `1.3.10`

**Comment by jijojosephk** [comment]
`package.json` is not updated :smile:

**Comment by thomasgl-orange** [comment]
@jijojosephk, see my previous comment, I went with 1.3.9 because there was no 1.3.9 release yet, so why would you skip this version number and jump straight to 1.3.10?
I can amend latest commit and set 1.3.10 everywhere, I would just like to understand why.

**Comment by jijojosephk** [comment]
> @jijojosephk, see my previous comment, I went with 1.3.9 because there was no 1.3.9 release yet, so why would you skip this version number and jump straight to 1.3.10? I can amend latest commit and set 1.3.10 everywhere, I would just like to understand why.

There is an unreleased version. I guess your fork is not updated. Can you sync the fork? @IsmaelMartinez had done some changes and bumped the version to `1.3.9` a few days ago.

**Comment by jijojosephk** [comment]
Thanks @thomasgl-orange!

**Comment by thomasgl-orange** [comment]
Great, thank you for the review and merge!

**Comment by jijojosephk** [comment]
`Pre-release`'d. You're the new superhero! :smile: Thanks a ton for your contribution.

**Comment by jijojosephk** [comment]
If you don't mind, you can join the chat room of the repo here: https://matrix.to/#/#teams-for-linux_community:gitter.im

**Comment by jijojosephk** [comment]
@thomasgl-orange there was a glitch in the code, I fixed in the latest PR. You may evaluate it for your reference.

**Comment by thomasgl-orange** [comment]
Huh, right, good catch.

**Pull request merged**